### PR TITLE
Explicit Decline of offers

### PR DESCRIPTION
**Pull request by Himavanth**
As per the documentation of mesos, if some offers are accepted, the others in the same lot should implicitly get declined.
But this is not happening in all cases and the offers get blocked as "outstanding offers" as seen in the mesos UI.
Expicitly declining them should do no harm afaics

### CODE REVIEW

File: src/main/scala/com/adobe/api/platform/runtime/mesos/MesosClient.scala [hunk 1]
```diff
@@ -347,7 +347,18 @@ trait MesosClientActor extends Actor with ActorLogging with MesosClientConnectio
...
+      val offerIds = asScalaBuffer(event.getOffersList).map(offer => offer.getId).filter(!matchedTasks.contains(_))
 
+      val declineCall = Call.newBuilder
+        .setFrameworkId(frameworkID)
+        .setType(Call.Type.DECLINE)
+        .setDecline(Call.Decline.newBuilder
+          .addAllOfferIds(seqAsJavaList(offerIds)))
+        .build;
+
+      execInternal(declineCall)
```
Comment (tysonnorris) [inline src/main/scala/com/adobe/api/platform/runtime/mesos/MesosClient.scala:360]: I think this future is not monitored for success?

Comment (tysonnorris) [reply src/main/scala/com/adobe/api/platform/runtime/mesos/MesosClient.scala:360]: Please add a test?

**Review by tysonnorris — CHANGES_REQUESTED** [review]

**Comment by tysonnorris** [comment]
Docs say that "accept" call will implicitly decline unused offer resources - I was under the impression (wrong?) that unaccepted offers will also be implicitly declined during accept processing. Can you verify that offers end up in a bad state in this case?

**Comment by Himavanth** [comment]
@tysonnorris Yes the docs do say that. In fact i was under the same impression and was puzzled to see outstanding offers in mesos UI while the sampleHAFramework was not getting the offers of unused resources. 
I don't see the outstanding offers anymore if i decline the offers first and then send the accept.

Though this is consistently reproducible with the sampleHAFramework, i am unsure why we would not see this in any of OW deployments.
Do you think there is something else at work here or is it safe to decline first?

outstanding_offers
unused_agents

**Comment by tysonnorris** [comment]
Is this because the sample framework shuts down without a decline? (and in OW maybe it doesn't do that). 
Also, is there any difference in declining after the accept? (if not, let's do that?)
Also, please check that there is handling of all the future completions, this may not be done currently I think @chetanmeh  pointed this out.
Thanks!

**Comment by tysonnorris** [comment]
included same functionality in #38

**Pull request closed without merging**